### ISSUE TITLE
[NPCs] Casters not moving when interupted

### ISSUE THREAD

**Issue by Imbarawr** [dalaranwow/dalaran-wow#4929]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)
When interupting Ragefire Shamans in Ragefire Chasm they stand at range waiting for the interupt CD to go away and then continues to cast spells. They do not move in melee range. Also unsure if we observed any melee attack performed by the Ragefire Shamans. We used Wind Shear as interupt.

Could be related to [4834](https://github.com/dalaranwow/dalaran-wow/issues/4834) and [3456](https://github.com/dalaranwow/dalaran-wow/issues/3456)

**Expected behaviour**: Tell us what should happen instead
When interupted by Wind Shear the Ragefire Shaman should move into melee range.

**Steps to reproduce the problem**:

1.  Go to Ragefire Chasm
2.  While at range interupt Ragefire Shaman with ranged interupt, I.E. Wind Shear.
3.  Observe as Ragefire Shaman continous to stay at range and not move into melee.

**Include proofs for this behaviour**
N/A

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**
Ragefire Shaman - https://wowgaming.altervista.org/aowow/?npc=11319

**Include Screenshots from the issue if necessary**
N/A

**Comment by TeridanDW** [comment]
Hello and thank you for the report @Imbarawr.
I can confirm the issue on PTR.
I've check the creature's SAI and it already has CMC flag.
@Rushor this must be a general issue with all creatures who use this flag, their behaviour in combat is correct but once interupted they should try to move in melee and autoattack as staying still and waiting for their spell/s to be avaliable again is not blizzlike.

**Comment by TeridanDW** [comment]
Creatures that share the same issue:
 - Burning Blade Apprentice - https://wowgaming.altervista.org/aowow/?npc=3198
 - Bristleback Geomancer - https://wowgaming.altervista.org/aowow/?npc=3263

**Comment by Rushor** [comment]
Hm yeah, then it must be a general problem with cmc flags

**Comment by Imbarawr** [comment]
More of the same issue:

Nak - https://wowgaming.altervista.org/aowow/?npc=3434

**Comment by Aridam** [comment]
Confirming this -- I'd say it's pretty much all Casters (and even Ranged Mobs -- i.e. those that primarily shoot you) who suffer from this, so if you do anything to disrupt their actions (like Interrupting Spells), they all just stay put; only LoS-ing seems to prove as somewhat of a workaround against this.

**Comment by TeridanDW** [comment]
After few tests on PTR the solution doesn't seem to work properly as:
 - creature do move after interupt for a moment, but if player runs away after a moment creature stops to move and again waits for school lockout to end.
![image](https://user-images.githubusercontent.com/6797611/32488499-7f14ab2c-c3b5-11e7-9d7a-64b0190c32db.png)

 - this commit seems to create a new issue: creatures with spell flag 64 now move with 1 step at a time everytime between their cast ends and new one starts.

**Comment by Aridam** [comment]
Yes that's pretty much what happens now -- they stop once you start getting away, and just stay there much like the original issue, furthermore they do indeed move a step at random times when they begin/end a cast.

There is also another thing I've noticed -- if you get to Melee Range and/or get out of it they'll move a step as if they're about to fight you in Melee just the same, however, _that_ does _not_ stop whatever ongoing cast they've got.

**Comment by Aridam** [comment]
By the way, reposting for another update: There also seems to be an issue with Mobs stuck on a waypoint moving rather than following/fighting you, that they just cast and keep casting _while moving to their waypoint_. Just happened in Azjol'Nerub with the starting Mobs that Cast -- one decided to go on with their waypoints in the middle of fighting and didn't even stop to cast or stop casting while moving.

**Comment by robinsch** [comment]
Above mentioned thing is not related to this issue.

**Comment by Aridam** [comment]
Pardon. Should I proceed to make (a) new report(s) for the above-stated then?

**Comment by robinsch** [comment]
Yeah sure, go ahead.
